drake/systems/framework/value.h

```diff
@@ -148,5 +157,18 @@ class VectorValue : public Value<VectorBase<T>*> {
   std::unique_ptr<VectorBase<T>> owned_value_;
 };
 
+// The following explicit instantiations are in libDrakeSystemFramework.
+// It is necessary to explicitly instantiate, and export, any implementation
+// of AbstractValue that would otherwise be implicitly instantiated elsewhere
+// in the library. If you do not create and export an explicit instantiation,
+// the Mach-O linker will generate a hidden typeinfo symbol in
+// libDrakeSystemFramework, and another hidden symbol in other libraries. These
+// typeinfos will have identical contents, but will not be identical for
+// the purpose of dynamic_cast, which will thwart unerasure of AbstractValue.
+
+
+extern template class DRAKESYSTEMFRAMEWORK_EXPORT Value<VectorBase<double>*>;
+extern template class DRAKESYSTEMFRAMEWORK_EXPORT VectorValue<double>;
```

Comment:
However, putting the export macro with the instantiation is an warning in GCC that is covered by `-Werror=all`:
```
warning: type attributes ignored after type is already defined [-Wattributes]
 template class DRAKESYSTEMFRAMEWORK_EXPORT Value<VectorBase<double>*>;
                                                       ^
```